docs/getting-started/dev-docker-setup.md

```diff
@@ -10,63 +10,79 @@ has_toc: true
-- **Pip**
-  + You should have pip installed as part of your python installation. If it is missing you need to install it [Installation instructions](https://pip.pypa.io/en/stable/installing/)
-- **Node 14 and npm**
-  + We recommend installing [nvm](https://github.com/nvm-sh/nvm#about) to manage node and npm versions.
+  + Check installation with `docker --version`
+- **Simple Python Version Management (Pyenv)**
+  + [Installation instructions](https://github.com/pyenv/pyenv#installation)
+  + Check installation with `pyenv --version`
+- **Node Version Manager (NVM)**
+  + [Installation instructions](https://github.com/nvm-sh/nvm#installing-and-updating)
+  + Check installation with `nvm --version`
   + NOTE: if after nvm installation you are getting `nvm: not found` you can use the following [troubleshooting tips](https://github.com/nvm-sh/nvm#troubleshooting-on-linux)
 - **Cairo**
   + [Installation instructions](https://invenio-formatter.readthedocs.io/en/latest/installation.html)
 
 ## Steps
 
-1. Install [invenio-cli](https://invenio-cli.readthedocs.io/en/latest/)
+1. Get the local copy of the UltraViolet source code:
   ```sh
-  pip install invenio-cli
+  git clone https://github.com/nyudlts/ultraviolet.git && cd ultraviolet
   ```
 
-  IMPORTANT: Check to make sure you have the latest version of Invenio-CLI tool. To check your version run `invenio-cli --version`. Versions are released about once per month, so if it's out of date, run `pip install --upgrade invenio-cli`
+2. Load correct python version in current environment. (Check result with `python --version`)
+  ```sh
+  pyenv install --skip-existing
+  ```
 
-2. Get the local copy of the UltraViolet source code:
+3. Load correct node version in current environment. (Check result with `node --version`)
   ```sh
-  git clone https://github.com/nyudlts/ultraviolet.git && cd ultraviolet
+  nvm install
   ```
 
-3. Create an `.invenio.private` file which is used by the invenio-cli tool
+4. Install and/or update pip and [invenio-cli](https://invenio-cli.readthedocs.io/en/latest/)
+  ```sh
+  pip install --upgrade pip
+  pip install --upgrade invenio-cli
+  ```
+
+5. Check Invenio's requirements
+  ```sh
+  invenio-cli check-requirements
+  ```
+
+6. Create an `.invenio.private` file which is used by the invenio-cli tool
   ```sh
   touch .invenio.private
   echo -e "[cli]\nproject_dir = "$PWD"\nservices_setup = False" >> .invenio.private
   ```
 
-4. Add .env file which will contain FLASK_SECRET_KEY
+7. Add .env file which will contain FLASK_SECRET_KEY
   ```sh
   touch .env
   echo FLASK_SECRET_KEY="some random value" > .env
   ```
 
-5. Build application python environment and web assets (make sure that you use node 14; check by running `node -v` and switch if you need to by running `nvm use 14`)
+8. Reset and build application python environment and web assets
   ```sh
+  pipenv --rm
```

Comment:
I am a bit lost here. Why do we need to delete the environment we may have created?

Reply:
to make sure we do not have old modules installed. We ran into some issues with that when working on POC

Reply:
OK. I lack the context to determine whether or not this is a proper solution or a hack, but I believe you